src/main/java/org/klodnicki/service/AccountService.java

```diff
@@ -47,13 +47,114 @@ private List<Account> getAllAccountsFromDatabase() {
+        }
+
+        if (parameterToModify.equalsIgnoreCase(SortOptionAccount.PESEL.getSortName()) && newData.length() != 11) {
+            throw new IllegalArgumentException("Pesel must have 11 digits.");
+
+        }
+
+        accountRepository.mergeAccount(peselAccountToModify, prepareParameterToDatabase(parameterToModify), newData);
+    }
+
+    private String prepareParameterToDatabase(String parameter) {
+
+        for (SortOptionAccount sortOption : SortOptionAccount.values()) {
+            if (sortOption.getSortName().equalsIgnoreCase(parameter)) {
+                parameter = sortOption.getHqlParameter();
+            }
+        }
+        return parameter;
+    }
+
+    public List<String> sortOptionNames() {
+        List<String> namesList = new ArrayList<>();
+        for (SortOptionAccount sortOption : SortOptionAccount.values()) {
+            namesList.add(sortOption.getSortName());
+        }
+        return namesList;
+    }
+
+    public List<String> prepareListOfAccountsByFirstNameAndLastName(String firstName, String lastName)
+            throws NotFoundInDatabaseException {
+        List<Account> accountsByFirstNameAndLastName = accountRepository.findAccountsByFirstNameAndLastName
+                (firstName, lastName);
+        List<String> results = new ArrayList<>();
+
+        if (accountsByFirstNameAndLastName.isEmpty()) {
+            throw new NotFoundInDatabaseException(Account.class);
+        }
+
+        for (Account account : accountsByFirstNameAndLastName) {
+            results.add(account.toString());
+        }
+        return results;
+    }
+
+    public List<String> prepareAllAccountsOrderByParameter(String parameter) {
+        List<Account> accountsOrderByParameter = accountRepository.findAllAccountsOrderByParameter
+                (prepareParameterToDatabase(parameter));
+        List<String> results = new ArrayList<>();
+
+        for (int i = 0; i < accountsOrderByParameter.size(); i++) {
+            results.add(accountsOrderByParameter.get(i).toString());
+        }
+        return results;
+    }
+
+    private boolean peselExist(String pesel, String firstName, String lastName) {
+        List<Account> accountList = accountRepository.findAccountsByFirstNameAndLastName(firstName, lastName);
+
+        for (int i = 0; i < accountList.size(); i++) {
+            if (pesel.equalsIgnoreCase(accountList.get(i).getPesel())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public List<String> prepareAccountByPeselWithBooks(String firstName, String lastName, String peselAccount)
+            throws NotFoundInDatabaseException {
+        List<String> results = new ArrayList<>();
+        List<Account> preparedList = accountRepository.findAccountsByFirstNameAndLastName(firstName, lastName);
+
+        for (int i = 0; i < preparedList.size(); i++) {
+            if (peselExist(peselAccount, firstName, lastName)) {
```

Comment:
This will always return true. This method is more like `peselExistOnList` which describes it more and what it does. I think it shouldn't be like that, because if you'll have i.e. 20 people named Jan Kowalski you'll print all 20 records of them + all their books they have. And what is described in the card is to display only one record and their books.

Reply:
you were right ! I was referring not to appropriate method while taking data from database. Now it works fine!